numpy/_core/src/multiarray/stringdtype/dtype.c

```diff
@@ -185,6 +185,30 @@ _eq_comparison(int scoerce, int ocoerce, PyObject *sna, PyObject *ona)
     return na_eq_cmp(sna, ona);
 }
 
+// Currently this can only return 1 or -1, the latter indicating that the
+// error indicator is set. Pass in out_na if you want to figure out which
+// na is valid.
+NPY_NO_EXPORT int
+stringdtype_compatible_na(PyObject *na1, PyObject *na2, PyObject **out_na) {
+    if ((na1 != NULL) && (na2 != NULL)) {
+        int na_eq = na_eq_cmp(na1, na2);
+
+        if (na_eq < 0) {
+            return -1;
+        }
+        else if (na_eq == 0) {
+            PyErr_Format(PyExc_TypeError,
+                         "Cannot find a compatible null string value for "
+                         "null strings '%R' and '%R'", na1, na2);
+            return -1;
+        }
+    }
+    if (out_na != NULL) {
+        *out_na = na1 ? na1 : na2;
+    }
+    return 1;
```

Comment:
Shouldn't this be `return 0`? (I.e., success.) I don't think any comparison other than `== -1` is done, so it doesn't affect anything, but 0 seems more logical.